"Error reading subtitle timings, file seems to be corrupted"

Operating System Version:
OS X 10.12.6

Popcorn Time Version:
Version 0.3.10

Download date:
May 9, 2019

#### Expected Behaviour
I simply wanted to load some subtitles from the in-app droplist.

#### Actual Behaviour
The app freezes for a moment, and then displays a message in a red box that reads:

"Error reading subtitle timings, file seems to be corrupted
Try another subtitle or simply drop one in the player"

It happens with all subtitles in any language.

#### Steps to reproduce the behaviour
1. Simply start playing any given show or movie.
2. Once it starts playing, try picking any language from the add subtitles option.
3. The screen will freeze for a moment, and the error will appear afterwards. The show or movie will also continue playing, but without any subtitles. The keyboard shortcuts for stuff like faster or slower playing or changing the timing for subtitles will also stop working.

![error](https://user-images.githubusercontent.com/28882527/58390665-07bdb180-7ff8-11e9-94ff-426d173bf078.jpg)

Harry Potter and the Philosopher's Stone with Greek subs gives me also the error for subtitle, but then it still shows it. Spacebar doesn't work but P does also I can use G and H for timings

![greek potter](https://user-images.githubusercontent.com/5569980/96933109-cdd82f80-14b7-11eb-913f-870342f9ff56.png)

When using my external (SMPlayer) player  I dont get the error and subs just play normal too.

Console output gives:
`videojshooks.js:334 Error reading subtitles timing, file seems corrupted TypeError: Cannot read property 'split' of undefined
    at subObj.vjs.TextTrack.parseCueTime (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/node_modules/video.js/dist/video-js/video.dev.js:8291:17)
    at subObj.vjs.TextTrack.parseCues (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/node_modules/video.js/dist/video-js/video.dev.js:8248:28)
    at vjsBind (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/videojshooks.js:332:23)
    at parse (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/videojshooks.js:297:17)
    at decode (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/videojshooks.js:326:17)
    at chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/videojshooks.js:362:17
    at Request._callback (chrome-extension://hecfofbbdfadifpemejbbdcjmfmboohj/src/app/vendor/videojshooks.js:132:25)
    at Request.self.callback (C:\Users\Mike\AppData\Local\Popcorn-Time\node_modules\request\request.js:185:22)
    at Request.emit (events.js:316:20)
    at Request.<anonymous> (C:\Users\Mike\AppData\Local\Popcorn-Time\node_modules\request\request.js:1161:10)`

--- a/src/app/lib/views/browser/list.js
+++ b/src/app/lib/views/browser/list.js
@@ -261,7 +261,7 @@
 
             Mousetrap.bind('i', function () {
                 if ((App.PlayerView === undefined || App.PlayerView.isDestroyed) && $('#player').children().length <= 0) {
-                    $('.about').click();
+                    $('#filterbar-about').click();
                 }
             }, 'keydown');
 

--- a/src/app/lib/views/browser/list.js
+++ b/src/app/lib/views/browser/list.js
@@ -255,7 +255,7 @@
 
             Mousetrap.bind(['`', 'b'], function () {
                 if ((App.PlayerView === undefined || App.PlayerView.isDestroyed) && $('#about-container').children().length <= 0 && $('#player').children().length <= 0) {
-                    $('.favorites').click();
+                    $('#filterbar-favorites').click();
                 }
             }, 'keydown');
 

--- a/src/app/lib/views/movie_detail.js
+++ b/src/app/lib/views/movie_detail.js
@@ -61,7 +61,7 @@
 
       //If a child was added above this view
       App.vent.on('viewstack:push', function() {
-        if (_.last(App.ViewStack) !== _this.className) {
+        if (_.last(App.ViewStack) !== _this.className && _.last(App.ViewStack) !== 'notificationWrapper') {
           _this.unbindKeyboardShortcuts();
         }
       });

--- a/src/app/lib/views/player/loading.js
+++ b/src/app/lib/views/player/loading.js
@@ -86,7 +86,7 @@
 
       //If a child was added above this view
       App.vent.on('viewstack:push', function() {
-        if (_.last(App.ViewStack) !== that.className) {
+        if (_.last(App.ViewStack) !== that.className && _.last(App.ViewStack) !== 'notificationWrapper') {
           that.unbindKeyboardShortcuts();
         }
       });

--- a/src/app/lib/views/player/player.js
+++ b/src/app/lib/views/player/player.js
@@ -1,6 +1,7 @@
 (function (App) {
     'use strict';
 
+    var _this;
     var Player = Marionette.View.extend({
         template: '#player-tpl',
         className: 'player',
@@ -30,6 +31,7 @@
         },
 
         initialize: function () {
+            _this = this;
             this.listenTo(this.model, 'change:downloadSpeed', this.updateDownloadSpeed);
             this.listenTo(this.model, 'change:uploadSpeed', this.updateUploadSpeed);
             this.listenTo(this.model, 'change:active_peers', this.updateActivePeers);
@@ -43,6 +45,13 @@
             this.firstPlay = true;
 
             this.boundedMouseScroll = this.mouseScroll.bind(this);
+
+            //If a child was removed from above this view
+            App.vent.on('viewstack:pop', function() {
+              if (_.last(App.ViewStack) === 'app-overlay') {
+                _this.bindKeyboardShortcuts();
+              }
+            });
         },
 
         isMovie: function () {

--- a/src/app/lib/views/player/player.js
+++ b/src/app/lib/views/player/player.js
@@ -763,10 +763,6 @@
                 that.toggleMute();
             }, 'keydown');
 
-            Mousetrap.bind(['u', 'U'], function (e) {
-                that.displayStreamURL();
-            }, 'keydown');
-
             Mousetrap.bind('j', function (e) {
                 that.adjustPlaybackRate(-0.1, true);
             }, 'keydown');
@@ -876,8 +872,6 @@
 
             Mousetrap.unbind(['m', 'M']);
 
-            Mousetrap.unbind(['u', 'U']);
-
             Mousetrap.unbind(['j', 'shift+j', 'ctrl+j']);
 
             Mousetrap.unbind(['k', 'shift+k', 'ctrl+k']);
@@ -973,12 +967,6 @@
 
         moveSubtitles: function (e) {
             AdvSettings.set('playerSubPosition', $('.vjs-text-track').css('top'));
-        },
-
-        displayStreamURL: function () {
-            var clipboard = nw.Clipboard.get();
-            clipboard.set($('#video_player video').attr('src'), 'text');
-            this.displayOverlayMsg(i18n.__('URL of this stream was copied to the clipboard'));
         },
 
         adjustSubtitleOffset: function (s) {

--- a/src/app/lib/views/show_detail.js
+++ b/src/app/lib/views/show_detail.js
@@ -74,7 +74,7 @@
 
             //If a child was added above this view
             App.vent.on('viewstack:push', function () {
-                if (_.last(App.ViewStack) !== _this.className) {
+                if (_.last(App.ViewStack) !== _this.className && _.last(App.ViewStack) !== 'notificationWrapper') {
                     _this.unbindKeyboardShortcuts();
                 }
             });

--- a/src/app/templates/keyboard.tpl
+++ b/src/app/templates/keyboard.tpl
@@ -257,10 +257,6 @@
                         <td><%= i18n.__('Set playback rate to %s', '4x') %></td>
                     </tr>
                     <tr>
-                        <td><span class="key">u</span></td>
-                        <td><%= i18n.__("Show Stream URL") %></td>
-                    </tr>
-                    <tr>
                         <td><span class="key">1</span></td>
                         <td><%= i18n.__("Set player window to video resolution") %></td>
                     </tr>

